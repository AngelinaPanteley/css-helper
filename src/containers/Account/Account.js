import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';

import styles from './Account.scss';

import Icon from '../../components/UI/Icon/Icon';
import Dropdown from '../../components/UI/Dropdown/Dropdown';
import Auxiliary from '../../hoc/Auxiliary/Auxiliary';
import * as actions from '../../store/actions/index';

class Account extends Component {
  state = {
    isDropdownOpen: false,
  }

  onAccountClick = (e) => {
    e.preventDefault();
    this.toggleDropdown();
  }

  toggleDropdown = () => {
    this.setState(prevState => {
      return {
        isDropdownOpen: !prevState.isDropdownOpen
      }
    });
  }

  render() {
    let links = [
      {
        title: 'Log In',
        route: '/login',
      },
      {
        title: 'Create Account',
        route: '/auth',
      }
    ];

    if (this.props.isAuth) {
      let showBy = localStorage.getItem('showBy');
      if (!showBy) {
        showBy = 5;
      }
      links = [
        {
          title: 'My Savings',
          route: `/savings/${showBy}/1`,
        },
        {
          title: 'Logout',
          route: '',
          click: this.props.onLogout,
        }
      ];
    }

    return (
      <Auxiliary>
        <div className={styles.Account}>
          {
            links.map((link) => {
              return (
                <span className={styles.Account_Link_Wrapper} key={link.route}>
                  <Auxiliary>
                    {
                      link.route
                        ?
                        <Link to={link.route}
                          className={styles.Account_Link}
                          onClick={link.click}>
                          {link.title}
                        </Link>
                        :
                        <a className={styles.Account_Link}
                          onClick={(e) => {
                            e.preventDefault();
                            link.click()
                          }}>
                          {link.title}
                        </a>
                    }
                  </Auxiliary>
                </span>
              )
            })
          }
        </div>
        <div className={styles.Account_Mobile}>
          <a href="" onClick={this.onAccountClick}>
            <Icon icon='account' />
          </a>
          {
            <Dropdown isOpen={this.state.isDropdownOpen}
              links={links} right
              onClose={this.toggleDropdown} />
          }
        </div>
      </Auxiliary>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    editorNames: state.editors.names,
    isAuth: !!state.auth.userId,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onLogout: () => dispatch(actions.logout()),
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Account));