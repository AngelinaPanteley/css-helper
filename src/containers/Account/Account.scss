@import '../../variables';

.Account {
  @include mobile {
    display: none;
  }

  &_Mobile {
    display: none;

    @include mobile {
      display: block;
    }
  }

  &_Link {
    text-decoration: none;

    &_Wrapper {
      margin: 0 10px;

      &:last-child {
        margin-right: 0;
      }

      &:first-child {
        margin-left: 0;
      }
    }
  }
}