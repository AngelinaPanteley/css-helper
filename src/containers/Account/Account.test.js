"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import { MemoryRouter, Link } from "react-router-dom";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Account from './Account.js';
import reducer from '../../store/reducers/index';

configure({ adapter: new Adapter() });

const store = createStore(reducer);

describe('>>>Account', () => {

  //SNAPSHOTS

  const props = {
    onLogout: () => { },
  }

  const accountComponent = <Provider store={store}>
    <MemoryRouter>
      <Account {...props} />
    </MemoryRouter>
  </Provider>;

  const component = renderer.create(accountComponent);

  it('renders itself correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
