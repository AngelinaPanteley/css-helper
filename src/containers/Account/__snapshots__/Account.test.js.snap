// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Account renders itself correctly 1`] = `
Array [
  <div>
    <span>
      <a
        href="/login"
        onClick={[Function]}
      >
        Log In
      </a>
    </span>
    <span>
      <a
        href="/auth"
        onClick={[Function]}
      >
        Create Account
      </a>
    </span>
  </div>,
  <div>
    <a
      href=""
      onClick={[Function]}
    >
      <i
        className="undefined undefined"
      />
    </a>
    <div
      className="  "
    >
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Log In
          </a>
        </li>
        <li>
          <a
            href="/auth"
            onClick={[Function]}
          >
            Create Account
          </a>
        </li>
      </ul>
    </div>
  </div>,
]
`;
